Src/AutoMoq/AutoMockPropertiesCommand.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using Moq;
+using Ploeh.AutoFixture.Kernel;
+
+namespace Ploeh.AutoFixture.AutoMoq
+{
+    /// <summary>
+    /// A command that populates all public writable properties/fields of a mock object with anonymous values.
+    /// </summary>
+    public class AutoMockPropertiesCommand : ISpecimenCommand
```

Comment:
> That it does, but it doesn't use mock for anything other than calling Execute with mock.Object. Why does it need to do that 'unwrapping', then?

The alternative to unwrapping `mock.Object` and passing it to an `AutoPropertiesCommand` would be to "decorate" the `Relay` with the `AutoPropertiesCommand` (since the `Relay` returns the type being mocked, not the `Mock<T>` type, and thus doesn't need unwrapping) as shown in my second code snippet. I'll explain why I'm not a big fan of wrapping the `Relay` below.

> Couldn't that specification just be added to an AutoPropertiesCommand instance?

It could. In the second code snippet above, for example, the customization wraps the `IRequestSpecification` in an `AutoPropertiesCommand`, which is then wrapped by a `PostProcessor`. But I personally believe avoiding all this wrapping and simply adding a new `ISpecimenCommand` to the existent post-processor looks cleaner.

I also think it's easier to understand what's going on in the first snippet ("the mock returned by the builder will be post-processed in 3 stages: first we'll setup its virtual methods, then stub its properties, and then populate them") than in the second snippet.

Since the first code snippet also keeps all the "mock configuration" stages in one place, it should also be easier to maintain, e.g. change the auto-configuration algorithm.

> Couldn't you make it a private member of the Customization instead?

If the `Customization` defined the `ISpecificationRequest` for the `AutoPropertiesCommand`, wouldn't that be against the Single Responsibility Principle?